### PR TITLE
[vpp] Add support of p010 to rgb4 conversion

### PR DESCRIPTION
**Pull request by rkazants**

### CODE REVIEW

File: _studio/shared/src/libmfx_allocator_vaapi.cpp [hunk 1]
```diff
@@ -389,6 +389,17 @@ mfxStatus mfxDefaultAllocatorVAAPI::SetFrameData(const VAImage &va_image, mfxU32
         }
         else mfx_res = MFX_ERR_LOCK_MEMORY;
         break;
+    case VA_FOURCC_P010:
+        if (mfx_fourcc == MFX_FOURCC_P010)
+        {
+            ptr->PitchHigh = (mfxU16)(va_image.pitches[0] / (1 << 16));
```
Comment (lakulako) [inline _studio/shared/src/libmfx_allocator_vaapi.cpp]: '>>16' and '& 0xffff' could be more clear.
Better use static_cast<mfxU16>.
At 399 better ptr->U + sizeof(mfxU16)

Comment (rkazants) [reply _studio/shared/src/libmfx_allocator_vaapi.cpp]: The formula for pitch high is written the same as for other formats. So propose to stay the current one.
I can correct line 339 as you proposed.

Comment (lakulako) [reply _studio/shared/src/libmfx_allocator_vaapi.cpp]: Don't multiply drawbacks.

Comment (rkazants) [reply _studio/shared/src/libmfx_allocator_vaapi.cpp]: Done.

**Pull request merged**